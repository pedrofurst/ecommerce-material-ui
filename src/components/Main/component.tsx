import { ReactElement } from 'react';
import { Container } from '@material-ui/core';
import { Switch, Route, BrowserRouter } from 'react-router-dom';
import CheckoutContainer from '@components/Checkout/container';
import ProductListContainer from '@components/ProductList/container';
import ProductDetailContainer from '@components/ProductDetail/container';
import CartContainer from '@components/Cart/container';
import useChildren from '@features/hooks/useChildren';
import useStyles from './styles';

type MainPropsType = {
  children: ReactElement;
};

function Main(props: MainPropsType) {
  const classes = useStyles();
  const { children } = props;
  const { getChildrenById } = useChildren(children);

  const topBar = getChildrenById('top-bar');

  return (
    <BrowserRouter>
      <div className={classes.mainContainer}>
        {topBar}
        <div className={classes.contentContainer}>
          <Container className={classes.content}>
            <Switch>
              <Route path="/" exact component={ProductListContainer} />
              <Route
                path="/product/:productId"
                component={ProductDetailContainer}
              />
              <Route path="/cart" component={CartContainer} />
              <Route path="/checkout" component={CheckoutContainer} />
            </Switch>
          </Container>
        </div>
      </div>
    </BrowserRouter>
  );
}

export default Main;
