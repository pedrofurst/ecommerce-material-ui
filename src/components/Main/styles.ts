import { makeStyles, Theme } from '@material-ui/core/styles';

export default makeStyles((theme: Theme) => ({
  mainContainer: {
    display: 'flex',
    height: '100vh',
    overflow: 'hidden',
    flexDirection: 'column',
    backgroundColor: '#F0E4E1',
  },
  contentContainer: {
    overflow: 'auto',
    padding: 24,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    flex: 1,
  },
  content: {
    padding: 0,
    flex: 1,
  },
}));
