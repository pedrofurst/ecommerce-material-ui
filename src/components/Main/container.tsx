import { useCallback, useEffect } from 'react';
import TopBarContainer from '@components/TopBar/container';
import useProductContext from '@features/providers/product/useProductContext';
import useCategoryContext from '@features/providers/category/useCategoryContext';
import useApi from '@features/hooks/useApi';
import Main from './component';

function MainContainer() {
  const { updateProducts } = useProductContext();
  const { updateCategories } = useCategoryContext();
  const { loadProducts, loadCategories } = useApi();

  const initializeProducts = useCallback(async () => {
    updateProducts(await loadProducts());
  }, [loadProducts, updateProducts]);

  const initializeCategories = useCallback(async () => {
    updateCategories(await loadCategories());
  }, [loadCategories, updateCategories]);

  useEffect(() => {
    initializeProducts();
  }, [initializeProducts]);

  useEffect(() => {
    initializeCategories();
  }, [initializeCategories]);

  return (
    <Main>
      <TopBarContainer id="top-bar" />
    </Main>
  );
}

export default MainContainer;
