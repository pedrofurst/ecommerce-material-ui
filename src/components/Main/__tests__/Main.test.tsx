import renderer from 'react-test-renderer';
import Main from '../component';

describe('<Main />', () => {
  it('Should renders main component properly', () => {
    const component = renderer.create(
      <Main>
        <div>children</div>
      </Main>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
