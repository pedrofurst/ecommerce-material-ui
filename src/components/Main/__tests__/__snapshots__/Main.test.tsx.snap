// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Should renders main component properly 1`] = `
<div
  className="makeStyles-mainContainer-1"
>
  <div
    className="makeStyles-contentContainer-2"
  >
    <div
      className="MuiContainer-root makeStyles-content-3 MuiContainer-maxWidthLg"
    >
      <div
        className="makeStyles-productListContainer-4"
      />
    </div>
  </div>
</div>
`;
